### PR TITLE
Species data standards

### PR DESCRIPTION
**Pull request by mauberti-bc**
## Links to Jira Tickets

- N/A
- Species standards describe what data (and formats) can be uploaded for a species. The new /standards endpoint just amalgamates lookup values for a species to display possible species measurement and marking body locations.

## Description of Changes

- New endpoint and service for retrieving data standards for a species.
- New page ("/standards") in the navbar for finding species standards
- New components for displaying species standards

## Testing Notes

- Navbar should include "Standards"
- After selecting a species, results should "Measurements" and "Marking Body Locations"
- Searching for a second species after an initial search should reload the component

### CODE REVIEW

**Comment by mauberti-bc** [comment]
Something's up with prettier: it updated a bunch of files in the API. Need to fix

File: app/src/features/standards/form/SpeciesStandardsForm.tsx
Comment (MacDeluca) [inline app/src/features/standards/form/SpeciesStandardsForm.tsx:1]: I think the code for this component can get pulled into SpeciesStandardPage

File: app/src/features/standards/view/SpeciesStandardsResults.tsx [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+  const resourcesDataLoader = useDataLoader((tsn: number) => biohubApi.standards.getSpeciesStandards(tsn));
+
+  useEffect(() => {
+    if (props.selectedSpecies) {
+      resourcesDataLoader.refresh(props.selectedSpecies.tsn);
+    }
+  }, [props.selectedSpecies?.tsn]);
+
+  if (!resourcesDataLoader.data) {
+    return <></>;
```
Comment (MacDeluca) [inline app/src/features/standards/view/SpeciesStandardsResults.tsx]: Empty state here would probably be nice

File: app/src/interfaces/useStandardsApi.interface.ts [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+ * @interface IGetSpeciesStandardsResponse
+ */
+export interface IGetSpeciesStandardsResponse {
+  tsn: number;
+  scientificName: string;
+  measurements: {
+    qualitative: CBQualitativeMeasurementTypeDefinition[];
+    quantitative: CBQuantitativeMeasurementTypeDefinition[];
+  };
+  markingBodyLocations: {
```
Comment (MacDeluca) [inline app/src/interfaces/useStandardsApi.interface.ts:20]: Maybe preferrable to query without the "asSelect" format. To be more inline with what a `markingBodyLocation` from critterbase is.

File: app/src/components/species/components/SpeciesAutoCompleteFormikField.tsx [hunk 3]
```diff
@@ -44,7 +44,7 @@ export const SpeciesAutoCompleteFormikField = (props: SpeciesAutoCompleteFormikF
       key={`${taxon?.[0].tsn}-formik-species-auto-complete`}
       formikFieldName={props.formikFieldName}
       disabled={props.disabled}
-      label={'Taxon'}
+      label={'Species'}
```
Comment (MacDeluca) [inline app/src/components/species/components/SpeciesAutoCompleteFormikField.tsx:47]: I think this component can also render non species taxons?

File: app/src/features/standards/SpeciesStandardsPage.tsx [hunk 4]
```diff
@@ -0,0 +1,26 @@
+import { Box, Container, Paper, Toolbar, Typography } from '@mui/material';
+import PageHeader from 'components/layout/PageHeader';
+import SpeciesStandardsForm from './form/SpeciesStandardsForm';
+
+const SpeciesStandardsPage = () => {
```
Comment (MacDeluca) [inline app/src/features/standards/SpeciesStandardsPage.tsx:14]: jsdocs

File: app/src/features/standards/view/SpeciesStandardsResults.tsx [hunk 5]
```diff
@@ -0,0 +1,101 @@
...
+import { useEffect, useState } from 'react';
+import MarkingBodyLocationStandardCard from './components/MarkingBodyLocationStandardCard';
+import MeasurementStandardCard from './components/MeasurementStandardCard';
+import SpeciesStandardsToolbar, { SpeciesStandardsViewEnum } from './components/SpeciesStandardsToolbar';
+
+interface ISpeciesStandardsResultsProps {
+  selectedSpecies?: ITaxonomy;
+}
+
+const SpeciesStandardsResults = (props: ISpeciesStandardsResultsProps) => {
```
Comment (MacDeluca) [inline app/src/features/standards/view/SpeciesStandardsResults.tsx:19]: jsdoc

File: app/src/features/standards/view/components/MarkingBodyLocationStandardCard.tsx [hunk 6]
```diff
@@ -0,0 +1,24 @@
+import { Paper, Typography } from '@mui/material';
+import { grey } from '@mui/material/colors';
+
+interface IMarkingBodyLocationStandardCard {
+  label: string;
+}
+
+const MarkingBodyLocationStandardCard = (props: IMarkingBodyLocationStandardCard) => {
```
Comment (MacDeluca) [inline app/src/features/standards/view/components/MarkingBodyLocationStandardCard.tsx:13]: jsdoc

File: app/src/features/standards/view/components/MeasurementStandardCard.tsx [hunk 7]
```diff
@@ -0,0 +1,66 @@
...
+import { useState } from 'react';
+
+interface IMeasurementStandardCard {
+  label: string;
+  description?: string;
+  options?: CBQualitativeOption[];
+  unit?: string;
+}
+
+const MeasurementStandardCard = (props: IMeasurementStandardCard) => {
```
Comment (MacDeluca) [inline app/src/features/standards/view/components/MeasurementStandardCard.tsx:20]: jsdoc

File: app/src/interfaces/useStandardsApi.interface.ts (hunk 2 above)
Comment (MacDeluca) [reply app/src/interfaces/useStandardsApi.interface.ts:20]: "asSelect" format simplifies the object down to id, key, value vs the original object shape

File: app/src/features/standards/view/SpeciesStandardsResults.tsx (hunk 1 above)
Comment (mauberti-bc) [reply app/src/features/standards/view/SpeciesStandardsResults.tsx]: What do you mean?

File: app/src/interfaces/useStandardsApi.interface.ts (hunk 2 above)
Comment (mauberti-bc) [reply app/src/interfaces/useStandardsApi.interface.ts:20]: Body position just needs the label, but might update in the future as the standards page evolves

File: app/src/features/standards/view/SpeciesStandardsResults.tsx (hunk 1 above)
Comment (MacDeluca) [reply app/src/features/standards/view/SpeciesStandardsResults.tsx]: In other components we have "empty state" which renders when no selection has occurred yet.
Like "No animal selected" or "No rows to show", if only we think that would be nice for this page.

File: app/src/hooks/api/useStandardsApi.test.ts [hunk 8]
```diff
@@ -0,0 +1,42 @@
...
+            value: 'left ear'
+          }
+        ],
+        measurements: {
+          quantitative: [{ measurement_name: 'body mass', measurement_desc: 'weight of the body' }],
+          qualitative: [{ measurement_name: 'life stage', measurement_desc: 'age class of the individual' }]
+        }
+      };
+
+      mock.onPost(`/api/standards/taxon/${mockTsn}`).reply(200, res);
```
Comment (NickPhura) [inline app/src/hooks/api/useStandardsApi.test.ts]: Should be `onGet`

**Comment by NickPhura** [comment]
Getting some reponse openapi errors

Opened the standards page.
Entered "Caribou"
Clicked "Caribou (Rangifer)"

The api request failed openapi validation:

```
{
  "name": "Internal Server Error",
  "status": 500,
  "message": "Invalid response for status code 200",
  "errors": [
    {
      "instancePath": "/response",
      "schemaPath": "#/properties/response/additionalProperties",
      "keyword": "additionalProperties",
      "params": { "additionalProperty": "tsn" },
      "message": "must NOT have additional properties"
    },
    {
      "instancePath": "/response/markingBodyLocations/0",
      "schemaPath": "#/properties/response/properties/markingBodyLocations/items/required",
      "keyword": "required",
      "params": { "missingProperty": "index" },
      "message": "must have required property 'index'"
    },
    {
      "instancePath": "/response/markingBodyLocations/0",
      "schemaPath": "#/properties/response/properties/markingBodyLocations/items/additionalProperties",
      "keyword": "additionalProperties",
      "params": { "additionalProperty": "id" },
      "message": "must NOT have additional properties"
    },
    {
      "instancePath": "/response/markingBodyLocations/1",
      "schemaPath": "#/properties/response/properties/markingBodyLocations/items/required",
      "keyword": "required",
      "params": { "missingProperty": "index" },
      "message": "must have required property 'index'"
    },
    {
      "instancePath": "/response/markingBodyLocations/1",
      "schemaPath": "#/properties/response/properties/markingBodyLocations/items/additionalProperties",
      "keyword": "additionalProperties",
      "params": { "additionalProperty": "id" },
      "message": "must NOT have additional properties"
    }
  ]
}
```
------------------

A few other random nits:

`nullable: false` is redundant (by default everything is `nullable: false`)

**Review by NickPhura — APPROVED** [review]
LGTM
